pinot-core/src/main/java/org/apache/pinot/core/util/IngestionUtils.java

```diff
@@ -114,4 +122,68 @@ public static Long extractTimeValue(Comparable time) {
     }
     return null;
   }
+
+  /**
+   * Fetches the streamConfig from the given realtime table.
+   * First, the ingestionConfigs->stream->streamConfigs will be checked.
+   * If not found, the indexingConfig->streamConfigs will be checked (which is deprecated).
+   * @param tableConfig realtime table config
+   * @return streamConfigs map
+   */
+  public static Map<String, String> getStreamConfigsMap(TableConfig tableConfig) {
```

Comment:
one high level question: shall we keep this utils method in pinot-core or we should move it to pinot-spi or pinot-common?

Reply:
Moved it to pinot-spi, but I didn't understand why it should be moved?
Do we also want to move all methods from IngestionUtils into the IngestionConfigUtils in the future?

Reply:
IMO the methods for ingestion config backward-compatibility should be in spi, others can stay in core